MonstieWash/Assets/Scripts/MonsterAi/MonsterBrain.cs

```diff
@@ -0,0 +1,308 @@
+
+                //Apply positive chain amount to each active mood that is in the list of positive reactions.
+                activeMoods[targetMood] += chainAmount * Time.deltaTime;
+
+                if (Debug) print("Active Mood: " + moodData[i].MoodName + " positively influenced " + moodData[targetMood].MoodName + "by amount " + chainAmount * Time.deltaTime);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Accesses moodsl ist of negative chain reaction moods, applies a negative reduction to the listed moods based on the negative reaction strength
+    /// and how far the current mood is inbetween its lower and upper bounds.
+    /// </summary>
+    private void negativeChainReactions()
+    {
+        //Loop through active moods.
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            if (MoodData[i].NegativeReactionStrength == 0) continue; //Skip if reaction strenght is 0
+
+            //Determine positive strength of current mood. (How far it is between its lower and upper limit)
+            var percentageStrength = ((activeMoods[i] - moodData[i].MoodLowerLimit) * 100) / (moodData[i].MoodUpperLimit - moodData[i].MoodLowerLimit);
+            float chainAmount = (percentageStrength / 100) * MoodData[i].NegativeReactionStrength * designerSanityBuff; //numbers are incredibly small *10 makes it more reasonable for designers.
+
+            //Loop through list of negative reactions in mood.
+            for (int j = 0; j < moodData[i].NegativeChainReactions.Count; j++)
+            {
+                int targetMood = accessActiveMoodIndexByMoodType(moodData[i].NegativeChainReactions[j]);
+
+                //Apply negative chain amount to each active mood that is in the list of negative reactions.
+                activeMoods[targetMood] -= chainAmount * Time.deltaTime;
+
+                if (Debug) print("Active Mood: " + moodData[i].MoodName + " negatively influenced " + moodData[targetMood].MoodName + " by amount " + chainAmount * Time.deltaTime);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Calls other function maintain limit for each mood currently active.
+    /// </summary>
+    private void maintainLimits()
+    {
+        for (int i = 0; i < activeMoods.Count; i++)
+        {
+            maintainLimit(i);
+        }
+    }
+
+    /// <summary>
+    /// Checks if mood given by its current index in active list has gone over or under its limits and then corrects it back.
+    /// </summary>
+    /// <param name="moodInt"> Given index of desired mood in active moods.</param> 
+    private void maintainLimit(int moodInt)
+    {
+        //Check upper limit and set back if over.
+        if (activeMoods[moodInt] > moodData[moodInt].MoodUpperLimit) activeMoods[moodInt] = moodData[moodInt].MoodUpperLimit;
+        //Check lower limit and set back if under.
+        if (activeMoods[moodInt] < moodData[moodInt].MoodLowerLimit) activeMoods[moodInt] = moodData[moodInt].MoodLowerLimit;
```

Comment:
This same effect can be done using Mathf.Clamp if you want to simplify this.
```suggestion
        activeMoods[moodInt] = Mathf.Clamp(activeMoods[moodInt], moodData[moodInt].MoodLowerLimit, moodData[moodInt].MoodUpperLimit);
```